### ISSUE TITLE
Lamp cannot dim by TRIAC when connect with PZEM 004T

### ISSUE THREAD

**Issue by Kosaki01** [mandulaj/PZEM-004T-v30#123]
I have a triac-dimmer circuit like below with load is incandescent lamp when i connect the N and L to 2 terminals of lamp, they cannot dim and work at max power - maximum brightness althought i just made they dim about 50% of max brightness. 
![image](https://github.com/mandulaj/PZEM-004T-v30/assets/142150141/a748ce0b-79e9-43df-a979-2a895be9766f)

How can i fix this? Pls give me some advice for this. Thank you so much.

**Comment by mandulaj** [comment]
Are you attaching the PZEM after the triac? If so, please attach it before the triac! The measurement will be the same.

**Comment by Kosaki01** [comment]
![image](https://github.com/mandulaj/PZEM-004T-v30/assets/142150141/6b6e03ae-4fdf-43bb-9911-34def530e4ce)
Hi, you mean this right?

**Comment by mandulaj** [comment]
LOAD is your lamp right, and the Power Triac is connected in series with the AC Line.... Can you attach the PZEM here where the power comes in?
![329289937-a748ce0b-79e9-43df-a979-2a895be9766f](https://github.com/mandulaj/PZEM-004T-v30/assets/3600920/4ee53c5e-3ae3-4e44-89c5-8b2fbcf25d1f)

**Comment by mandulaj** [comment]
Ahh, but then if you are interested in the voltage, PZEM is not the right sensor to use! PZEM is good for measuring the power/energy! The AC input is a continuous sine wave, but the triac will chop it up (which is how it achieves the power modulation)

**Comment by Kosaki01** [comment]
Yeah the main reason i used pzem is for measuring power, but with the fault of voltage, they cannot give right value of power :(

**Comment by mandulaj** [comment]
But if you do as I told you and attach the PZEM before the Triac and put the current probe around one of the wires, the power measurement should be done correctly. The PZEM will take the correct integrated product of voltage and current over time.

**Comment by Kosaki01** [comment]
![image](https://github.com/mandulaj/PZEM-004T-v30/assets/142150141/7aa7f535-186c-430d-b8ac-cb9ebf4fc2a3)
Yeah i did it, and i compared pzem's value with VOM's value, they had difference power and voltage value when lamp is dimmed, and same value when lamp is at max brighness - max power comsumption.

**Comment by mandulaj** [comment]
With your multimeter, did you measure voltage and current independently? Because if yes, their product is not the actual power for a AC system. A dimmed light will not have a power factor of 1. The Voltage and Current are not in phase! Thus if you only look at the magnitude of the Voltage and Current and their product, you won't get the Real power (There is some reactive power) https://en.wikipedia.org/wiki/AC_power

**Comment by Kosaki01** [comment]
So if i measure with an oscilloscope, i will get the real power, right?
